src/transformers/convert_slow_tokenizer.py

```diff
@@ -625,14 +625,26 @@ def converted(self) -> Tokenizer:
         # control tokens are special
         # user defined symbols are not
         # both user and control tokens are AddedTokens
-        # Add user defined symbols (type == 4) from sentnecepiece (https://github.com/google/sentencepiece/blob/6225e08edb2577757163b3f5dbba4c0b670ef445/src/sentencepiece_model.proto#L299C29-L299C33)
+        # Add user defined symbols (type == 4) from sentencepiece (https://github.com/google/sentencepiece/blob/6225e08edb2577757163b3f5dbba4c0b670ef445/src/sentencepiece_model.proto#L299C29-L299C33)
 
         tokens_to_add = {
             id: AddedToken(token, normalized=False, special=special)
             for id, token, special in [
                 (id, p.piece, p.type == 3) for id, p in enumerate(self.proto.pieces) if p.type in [3, 4]
             ]
         }
+
+        if isinstance(tokenizer.model, BPE) and not getattr(tokenizer.model, "ignore_merges", False):
+            # To adhere to the BPE standard, if ignore_merges=False (default), we must
+            # remove all user-defined tokens that can be obtained via merges
```

Comment:
wow that's a bit weird and gonna be slow no slow 😅 can't we do this when create the merged (so create tokens_to_add an pass it)

Reply:
I was trying to find a way to do that (since this approach is kind of hacky) 👀 but I didn't see how to pass this information around without completely rewriting the `.tokenize` function.

Reply:
Why not pass it here: https://github.com/huggingface/transformers/blob/cc7c1ddb511f21239c7be1e32e05928829d2bfec/src/transformers/convert_slow_tokenizer.py#L68-L95

Reply:
I'm not sure I understand 👀 the selection of tokens to add occurs here:
https://github.com/huggingface/transformers/blob/654820751af3c0d82f6c77e49c679184ca888f27/src/transformers/convert_slow_tokenizer.py#L622-L646
meaning we would need to rewrite the `self.tokenizer(...)` function to return `tokenizer, tokens_to_add = ...`. Could be cleaner, but means the spm `.tokenizer` function will act differently to all others.

Reply:
Actually, looking at it again, it doesn't look so bad. Let me make that change 👍

Reply:
Ok!

Reply:
Refactor done ✅ https://github.com/huggingface/transformers/pull/32191/commits/8cea780cab933b1783129ba49f09fd3ab85346cb

Doing tests with llama and other spm converters now